openshift-s2i/s2i.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: s2i
+spec:
+  inputs:
+    resources:
+     - name: git-source
+       type: git
+    params:
+    - name: BUILDER_IMAGE
+      description: The location of the builder image.
+      default: quay.io/openshift-pipeline/kbaig_s2i
+    - name: S2I_IMAGE
+      description: The location of the s2i builder image.
+      default: docker.io/centos/nodejs-8-centos7
+    - name: PATH_CONTEXT
+      description: The location of the path to run s2i from.
+      default: .
+
+  outputs:
+    resources:
+    - name: image
+      type: image
+
+  steps:
+  - name: generate
+    image: ${inputs.params.BUILDER_IMAGE}
+    workingdir: /workspace/git-source
+    command: ['s2i', 'build', '${inputs.params.PATH_CONTEXT}', '${inputs.params.S2I_IMAGE}', '--as-dockerfile', '/gen-source/Dockerfile.gen']
+    volumeMounts:
+     - name: gen-source
+       mountPath: /gen-source
+  - name: build-and-push
+    image: gcr.io/kaniko-project/executor
```

Comment:
Let's use `buildah` here

Reply:
OK. I didn't use buildah because it requires privilege.

Reply:
Changed to buildah